### DISCUSSION TITLE
Threat model template using property type - Single item from a list - How to check at least one item is selected

### DISCUSSION THREAD

**Discussion by rsrinivasanhome** [simonec73/threatsmanager#76] [Q&A] [answered]
In my threat model template many item templates have drop downs with multiple values.  We have used the property type - "Single item selected from a list".
I want to make sure at least one value is selected.   Is it possible ?  I have tried rules but it did not work.

![image](https://github.com/simonec73/threatsmanager/assets/31912418/7ea49c3e-2964-4323-9f2f-8012c1fb8122)

**Comment by simonec73** [comment, marked as answer]
No, it's not possible. 

If you can accept a quality validation that highlights violations without enforcing anything, you could create a Quality Analyzer that would highlight when you have a field that has not been set. Please refer to https://github.com/simonec73/threatsmanager/tree/master/Sources/Extensions/ThreatsManager.Quality/Analyzers for examples on how to do that. For example, https://github.com/simonec73/threatsmanager/blob/master/Sources/Extensions/ThreatsManager.Quality/Analyzers/FlowBetweenDataStores.cs considers the validation failed as soon as you have a direct flow between Data Stores.
